dbms/src/Storages/DeltaMerge/SegmentReadTask.h

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Storages/DeltaMerge/Remote/DisaggTaskId.h>
+#include <Storages/DeltaMerge/Remote/Proto/remote.pb.h>
+#include <Storages/DeltaMerge/Remote/RNLocalPageCache_fwd.h>
+#include <Storages/DeltaMerge/RowKeyRange.h>
+#include <Storages/DeltaMerge/Segment.h>
+#include <Storages/KVStore/Types.h>
+
+namespace DB::DM
+{
+struct SegmentReadTask;
+using SegmentReadTaskPtr = std::shared_ptr<SegmentReadTask>;
+using SegmentReadTasks = std::list<SegmentReadTaskPtr>;
+
+struct RemoteSegmentID
+{
+    KeyspaceID keyspace_id;
+    StoreID store_id;
+    TableID physical_table_id;
+    UInt64 segment_id;
+
+    String toString() const
+    {
+        return fmt::format(
+            "RemoteSegmentID<store_id={} keyspace={} table_id={} segment_id={}>",
+            store_id,
+            keyspace_id,
+            physical_table_id,
+            segment_id);
+    }
+};
+
+struct ExtraRemoteSegmentInfo
+{
+    RemoteSegmentID remote_segment_id;
+    String store_address;
+    DisaggTaskId snapshot_id;
```

Comment:
Why this is called `snapshot_id`? It will be better to add more comments in this file.